### PR TITLE
feat: support shell scripts with no extension, fix perl shebang regex

### PR DESCRIPTION
**Pull request by tnyeanderson**
## Proposed Changes

1. Support sh/bash scripts which do not have an extension based on their shebang
2. Fix the perl shebang regex to only include shebangs at the beginning of the file

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by tnyeanderson** [comment]
Not sure if I added the CHANGELOG entry here since I'm unsure what the versioning scheme will be. Please correct me if I'm wrong.

File: CHANGELOG.md [hunk 1]
```diff
@@ -184,6 +184,10 @@ Note: Can be used with `oxsecurity/megalinter@beta` in your GitHub Action mega-l
   - [cfn-lint](https://github.com/aws-cloudformation/cfn-lint) from 1.15.1 to **1.15.2** on 2024-09-30
 <!-- linter-versions-end -->
 
+## [v8.0.1] - 2024-08-19
+
+- Support shell scripts with no extension and only support perl shebangs at the beginning of a file in <https://github.com/oxsecurity/megalinter/pull/4076>
+
```
Comment (echoix) [inline CHANGELOG.md:190]: Can you place that entry in the section that starts on line 7?

File: megalinter/descriptors/bash.megalinter-descriptor.yml [hunk 2]
```diff
@@ -7,10 +7,17 @@ descriptor_flavors:
   - c_cpp
   - formatters
 file_extensions:
+  - ""
```
Comment (echoix) [inline megalinter/descriptors/bash.megalinter-descriptor.yml:10]: What is the impact of this in projects?

File: CHANGELOG.md (hunk 1 above)
Comment (tnyeanderson) [reply CHANGELOG.md]: Thanks, done

File: megalinter/descriptors/bash.megalinter-descriptor.yml (hunk 2 above)
Comment (tnyeanderson) [reply megalinter/descriptors/bash.megalinter-descriptor.yml:10]: If the file without an extension does not include a matching shebang, my understanding is that there will be [no impact](https://github.com/oxsecurity/megalinter/blob/d5670fa4d09893729c81131fb6e71f15790e3770/megalinter/utils.py#L206-L208)

Comment (tnyeanderson) [reply megalinter/descriptors/bash.megalinter-descriptor.yml:10]: However, this is needed in order for the conditional logic to even reach the point where it checks for matches in `file_contains_regex`

**Review by echoix — APPROVED** [review]

**Pull request merged**